governance/pyth_staking_sdk/src/utils/pool.ts

```diff
@@ -0,0 +1,34 @@
+import { PublicKey } from "@solana/web3.js";
+
+import type { PoolDataAccount, PublisherData } from "../types";
+
+export const extractPublisherData = (
+  poolData: PoolDataAccount,
+): PublisherData => {
+  return poolData.publishers
+    .filter((publisher) => !publisher.equals(PublicKey.default))
+    .map((publisher, index) => ({
+      pubkey: publisher,
+      stakeAccount:
+        poolData.publisherStakeAccounts[index] === undefined ||
+        poolData.publisherStakeAccounts[index].equals(PublicKey.default)
+          ? null
```

Comment:
Minor preference but I suggest just using `undefined` instead of mixing undefined and null; the fact that javascript has both is a wart and it's very very easy to misuse them in subtle ways.  For all intents and purposes they are typically used interchangeably, so best to just standardize on one and always use it.  Due to the semantics of missing values, you generally can't avoid `undefined`, but `null` is generally only introduced in user code, so I generally suggest just standardizing on using `undefined` everywhere.

If you need a true ternary state, I will always recommend using the [fake sum types](https://www.typescriptlang.org/play/#code/C4TwDgpgBAIghsOUC8UDeAoK2oDc4A2ArhAFxQB2RAtgEYQBOGAvgNwYYRXVQDKiwCABVw0TDigA5APbB+cBoIAmAGiw4AMtLhKAlhQDmaiQFEGDaQ2M5eRAMZ2IAZycsOd6RSfA+A6KnEcGTlERQglcgAKAEoUAD4oSLQoUEhyeUERSAA6YPkwpSg4JygPLx9maOtsLR19AyjY5ASklNF0vyyIbNq9QyKSsu8oSuqoMwsGKMZJ8iIKAGsKaQB3CiaW5NSyXwRhUWyJywHSz28VKBnj0fVsWwdnJyilBDhyeEQNxK323cyD+6OFwnIbAC4vRAjaIsdgYbZ-fxQABKEGARAYFC6AB5ttIAGYIgDaCwgIHxbUg5IyEAAunFYRhQVAnAALVbUlCJbx7Dp7L6BbBOFa6YB2FlcvzZbaxAUSUrFaDUrq5WT5ZTkBio9EUKAAImCuwKAEJdew5Tg7AqEcrevUNVqMXrbYYTWbzfKnIrOgcjlMoJq0Y6AAYAQR1VwYUGkDnRmoiUAAJGhuYJshHmEG3ebLZ7rQD7ECnv6HTqg71wnhCCRyEmU90IXBsvhiBAM1mRm5GWdpARugRpAZIqz2X5ItSVSEFMoYtFouwhj2+wOh2yVtSx5LnYPZ3Ou15F9l+4Ph2vR+PfZEKBAVuNzJZIrqiOLpCzdTvdwve4flyf1+PAY8rTNtWUAAKwAAxQrOrBAA) pattern I shared with you earlier, it's far more explicit, easier to understand the intent, and less error-prone.